### DISCUSSION TITLE
Store previously used CSS so it can be reused during the next upload

### DISCUSSION THREAD

**Discussion by Stvad** [roam-garden/i#32] [Feature requests & Ideas] [answered]
Most of the time you'd iterate on your CSS some and then settle to some standard version that you'd update rarely. So in such a situation it's annoying to have to re-upload the same CSS each time.   
It'd be great if it were preserved and reused between upload sessions.

### LocalStorage
One option is to change the CSS field from being a file upload to being a code-entry field & store it to browser localstorage similar to how other upload settings work.  
The main problem here is that LocalStorage is somewhat transient and not a great thing to rely on for long-term persistent data

### Preserve the uploaded CSS file on a longer-term basis and store link in LocalStorage
This one has basically the same characteristics as a previous option but is probably more complicated infrastructurally

### Store more metadata about site on a backend
Including the link to latest uploaded CSS. This is probably a preferred long-term solution, but not sure when I get there

### Read CSS from inside Roam 
Another potentially interesting option is to support reading the CSS from some page in Roam. This would likely be required for #28

**Comment by Stvad** [comment, marked as answer]
So this is partially implemented with the new UI for upload - it's storing the CSS in local storage

**Reply by p3ob7o** [reply]
Is there a way to get the custom CSS from the roam.garden instance? After switching to a new laptop, I don't have access to the local storage on the previous one anymore.

**Reply by Stvad** [reply]
hmm, one way is to run a browser inspector on your garden and extract the css of the parts you're interested in.  

there is potentially another way to recover it if the version you're interested in has been published less then 30 days ago (it involves me manually tracking down temp files created during the build process)

**Reply by p3ob7o** [reply]
Thanks for the offer. In this case, it was easier to rewrite it (and improve it on the way), as it was added in early September and was anyway older than 30 days. I am not sure how the whole roam/garden configuration page works, but as it is a page in Roam, I think the expected behaviour would be to store whatever we type in it. I solved this by copy/pasting the custom CSS, keeping a copy in a block underneath the Submit button on that page.
